### PR TITLE
bug #4508 - removes tokens and fiat currencies unsupported by cryptoc…

### LINKED ISSUE #4508: Total value is shown as zero for some currencies and tokens because they are not supported in API
### User Story

As a user, I want to see only tokens for which price can be correctly fetched; and currencies only suitable for conversion;

### Description

*Type*: Bug
*Summary*: as we are using https://min-api.cryptocompare.com for fetching tokens price, there are several tokens presented in Status app, but not provided by https://min-api.cryptocompare.com/.
List of tokens:
- ROL
- Centra
- ATMChain

List of fiat currencies: 
BMD, BZD, CPL, GMD, GYD, KYD, LAK, LRD, LTL, MKD, MNT, NIO, SOS, SRD, UGH, YER

**Additionally:**
I found 20 currencies(ALL, AZN,  BSD, BYN, BAM,  BWP, KHR, CLP, SVC, GHC, GIP, GTQ, GGP, HNL, IRR, KGS, LBP, PAB, TRL, UZS) that are presented in cryptocompare API, but missing in Status app.
Probably there is another reason to exclude them from app. 
All currencies + list of requests/responses are listed [here](https://github.com/status-im/status-react/files/2053226/results_missing_curr.txt)

#### Expected behavior

*Scenario 1:*
**Status app:** price in USD
**API:** price in USD returned 

*Scenario 2:*
**Status app:** price in selected currency
**API:** price in selected currency returned

#### Actual behavior

*Scenario 1:*
**Status app:** total value in assets for these tokens will be always 0 (doesn't depend on currency)
API: Error: There is no data for the symbol {token_name}

*Scenario 2:*
**Status app:** total value in assets in selected currency will be always 0 (doesn't depend on tokens)
bmd

**API:** "There is no data for any of the toSymbols {currency_name} ."

### Reproduction

*Scenario 1:*
- Open https://min-api.cryptocompare.com/
- Select "Price"
- Check requests:
https://min-api.cryptocompare.com/data/price?fsym=ROL&tsyms=USD
https://min-api.cryptocompare.com/data/price?fsym=Centra&tsyms=USD
https://min-api.cryptocompare.com/data/price?fsym=ATMChain&tsyms=USD

*Scenario 2:*
- Open https://min-api.cryptocompare.com/
- Select "Price"
- Check requests listed [here](https://github.com/status-im/status-react/files/2052806/results.txt)

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*: 
Dropping these tokens\currencies will be the simplest solution.

### Additional Information
* Status version: 0.9.19 (6016) - nightly 29.05
* Operating System: Android,iOS

### PR DESCRIPTION
**Pull request by goranjovic**
fixes #4508 

### Summary:

Removes a number of tokens and currencies which are unsupported by cryptocompare. Includes a migration that sets the selected currency in Profile to USD if the previous currency was among the removed ones. Also includes a migration that deselects removed tokens from user's selection of tokens in Wallet.

"Removed" tokens are merely removed from the selection and then hidden from the list of all possible assets. They are not deleted from the code because we still want to show any previous token transactions in the history even if we can't tell their fiat value.

### Testing notes (optional):

Keep an eye on migrations. Specific cases are:
- when currency in Profile was one of the removed ones (should be set to USD by migration)
- when currency was not among removed ones (should remain as it was)
- when user had removed tokens in their selection on main Wallet screen (should be removed from both selection and Manage Assets)
- when user had transactions with said tokens in Transaction history (transactions should still appear)

### Steps to test:
- Open Wallet and check the Token lists
- Open Profile and check the fiat currencies lists

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/currency_settings/events.cljs [hunk 1]
```diff
@@ -7,4 +7,5 @@
  (fn [{:keys [db] :as cofx} [_ currency]]
    (let [settings     (get-in db [:account/account :settings])
          new-settings (assoc-in settings [:wallet :currency] currency)]
+     #_(js/alert new-settings)
```
Comment (jeluard) [inline src/status_im/ui/screens/currency_settings/events.cljs]: ?

Comment (jeluard) [inline src/status_im/ui/screens/currency_settings/events.cljs]: ?

**Review by jeluard — COMMENTED** [review]
Wondering if the migration complexity is worth it. We could make the code resilient to such selections and inform users they have to manually change to something else?

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/3RvrYb for branch PR-4538

<blockquote><div><strong><a href="https://i.diawi.com/3RvrYb">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: src/status_im/ui/screens/currency_settings/events.cljs (hunk 1 above)
Comment (goranjovic) [reply src/status_im/ui/screens/currency_settings/events.cljs]: crap, removed

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/yM4CDT for branch PR-4538

<blockquote><div><strong><a href="https://i.diawi.com/yM4CDT">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by goranjovic** [comment]
@jeluard Migration of settings (currently selected fiat and tokens) might be worth it or might not. Migration of transaction history definitely isn't worth it.

Informing the user is practically just as complex because you have to do that before the first time when a removed value is used, so not fun.

The other valid alternative is to leave those tokens and fiats in. I don't think it's too unusual for a token to be unsupported by some app somewhere and especially so for newer tokens. The same goes for currencies from small countries.

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/3w6CWJ for branch PR-4538

<blockquote><div><strong><a href="https://i.diawi.com/3w6CWJ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/n11dcP for branch PR-4538

<blockquote><div><strong><a href="https://i.diawi.com/n11dcP">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/syM5sW for branch PR-4538

<blockquote><div><strong><a href="https://i.diawi.com/syM5sW">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/4N8K3R for branch PR-4538

<blockquote><div><strong><a href="https://i.diawi.com/4N8K3R">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Review by cammellos — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

**Review by janherich — APPROVED** [review]

**Comment by asemiankevich** [comment]
`KYD` is still in the list of currencies, maybe it was missed occasionally

**Comment by asemiankevich** [comment]
looks good to me! KYD is not shown anymore. Migration looks fine (unsupported assets are not shown, unsupported currencies are replaced with USD when upgrading)

**Pull request merged**